### ISSUE TITLE
regression in ArrayTypeStyle

### ISSUE THREAD

**Issue by hgschmie** [checkstyle/checkstyle#10355]
```bash
$ cd /tmp
$ git clone git@github.com:jdbi/jdbi
$ cd jdbi
$ ./mvnw checkstyle:check
[... check is successful ...]
$ find . -name checkstyle-cachefile | xargs rm -f
$ ./mvnw -Ddep.checkstyle.version=8.44 checkstyle:check
[ERROR] /private/tmp/jdbi/core/src/main/java/org/jdbi/v3/core/mapper/MapEntryMapper.java:42:39: Array brackets at illegal position. [ArrayTypeStyle]
[ERROR] /private/tmp/jdbi/core/src/main/java/org/jdbi/v3/core/generic/GenericTypes.java:35:33: Array brackets at illegal position. [ArrayTypeStyle]
```
---

Expected: build still works fine with the ruleset we have been using for a long time.
Actual result: see above

[ERROR] /private/tmp/jdbi/core/src/main/java/org/jdbi/v3/core/mapper/MapEntryMapper.java:42:39: Array brackets at illegal position. [ArrayTypeStyle]
[ERROR] /private/tmp/jdbi/core/src/main/java/org/jdbi/v3/core/generic/GenericTypes.java:35:33: Array brackets at illegal position. [ArrayTypeStyle]

---

This is a regression from 8.43 to 8.44. The offending lines:

```java
        TypeVariable<Class<Map.Entry>>[] mapParams = Map.Entry.class.getTypeParameters();
```

```java
        TypeVariable<Class<Map>>[] mapParams = Map.class.getTypeParameters();
```

My guess is that the code gets confused by the generic types for the arrays.

**Comment by strkkk** [comment]
@nmancus1 looks like another regression)

**Comment by rnveach** [comment]
@hgschmie https://checkstyle.org/report_issue.html#How_to_report_a_bug.3F

https://github.com/jdbi/jdbi/blob/master/policy/src/main/resources/policy/checkstyle.xml#L39
https://github.com/jdbi/jdbi/blob/0420cc462f98565062bd577cf5d3a4d92a621060/core/src/main/java/org/jdbi/v3/core/mapper/MapEntryMapper.java#L42

**Comment by nmancus1** [comment]
This is the same regression that was reported in https://github.com/checkstyle/checkstyle/issues/10215. 

```bash

➜  src cat Test.java
import java.lang.reflect.TypeVariable;
import java.util.Map;

public class Test {
    TypeVariable<Class<Map.Entry>>[] mapParams1 = Map.Entry.class.getTypeParameters();
    TypeVariable<Class<Map>>[] mapParams2 = Map.class.getTypeParameters();
}

➜  src javac Test.java

➜  src cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <property name="haltOnException" value="true"/>
    <property name="severity" value="error"/>

    <module name="TreeWalker">
        <module name="ArrayTypeStyle">
        </module>
    </module>
</module>

```

No violations with 8.43, as reported by user:

```bash
➜  src java -jar checkstyle-8.43-all.jar -c config.xml Test.java                                          
Starting audit...
Audit done.
```

Regression with 8.44, as reported by user:

```bash
➜  src java -jar checkstyle-8.44-all.jar -c config.xml Test.java                                          
Starting audit...
[ERROR] /home/nick/IdeaProjects/Tester/src/Test.java:5:35: Array brackets at illegal position. [ArrayTypeStyle]
[ERROR] /home/nick/IdeaProjects/Tester/src/Test.java:6:29: Array brackets at illegal position. [ArrayTypeStyle]
Audit done.
Checkstyle ends with 2 errors.
```
Latest master:

```bash
➜  src java -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.45-SNAPSHOT-all.jar -c config.xml Test.java
Starting audit...
Audit done.

```
@hgschmie Thanks for your report. We have fixed this in https://github.com/checkstyle/checkstyle/pull/10217, and will be included in the 8.45 release at the end of this month.

**Comment by hgschmie** [comment]
@nmancus1  Thank you for the quick response. Looking forward to 8.45 release.
